controllers/cf/cf_pod_spec.go

```diff
@@ -0,0 +1,318 @@
+			currentPodSpec = currentPodSpec.DeepCopy()                                             // TODO?
+			factoryPodSpec := this.services.GetKubeFactory().CreateDeployment().Spec.Template.Spec // TODO Cache this?
+			targetPodSpec, err := SanitizeBasePodSpec(this.log, this.basePodSpec, currentPodSpec, &factoryPodSpec)
+			if err != nil {
+				if err.isConfigError {
+					this.log.Errorw("PodSpec field in spec.deployment.podSpecPreview is invalid", "error", err)
+					this.services.GetConditionManager().GetConfigurationErrorCondition().
+						TransitionInvalid(err.Error(), "spec.deployment.podSpecPreview")
+					// No need to transition to not ready, since we can just with the previous config
+					this.ctx.SetRequeueDelaySec(10)
+				} else {
+					this.log.Errorw("Conversion error", "error", err.conversionError)
+				}
+			} else {
+
+				this.targetPodSpec, err = ConvertToPodSpec(targetPodSpec)
+				if err != nil {
+					this.log.Errorw("Conversion error", "error", err.conversionError)
+				} else {
+					this.valid = true
+				}
+			}
+		}
+	}
+}
+
+func (this *PodSpecCF) Compare() bool {
+	if this.previousBasePodSpec != nil { // common.LogNillable does not work TODO find out why
+		this.log.Debugw("Obsevation #1", "this.previousBasePodSpec", this.previousBasePodSpec)
+	} else {
+		this.log.Debugw("Obsevation #1", "this.previousBasePodSpec", "<nil>")
+	}
+	this.log.Debugw("Obsevation #2", "this.basePodSpec", this.basePodSpec)
+	this.log.Debugw("Obsevation #3", "this.targetPodSpec", this.targetPodSpec)
+	return this.valid &&
+		// We're only comparing changes to the podSpecPreview, not the real pod spec,
+		// so we do not overwrite changes by the other CFs, which would cause a loop panic
+		(this.previousBasePodSpec == nil || !reflect.DeepEqual(this.basePodSpec, this.previousBasePodSpec))
+}
+
+func (this *PodSpecCF) Respond() {
+
+	if entry, exists := this.svcResourceCache.Get(resources.RC_KEY_DEPLOYMENT); exists {
+		entry.ApplyPatch(func(value interface{}) interface{} {
+			deployment := value.(*apps.Deployment).DeepCopy()
+
+			deployment.Spec.Template.Spec = *this.targetPodSpec
+
+			this.previousBasePodSpec = this.basePodSpec
+
+			return deployment
+		})
+	}
+}
+
+func (this *PodSpecCF) Cleanup() bool {
+	// No cleanup
+	return true
+}
+
+/*
+Not allowed:
+
+- affinity [alternative exists]
+- containers[*]. [only a single container without a name]
+  - containers.args [reserved]
+  - containers.command [reserved]
+  - containers.env [alternative exists]
+  - containers.image [reserved]
+  - containers.imagePullPolicy [alternative exists]
+  - containers.name [reserved]
+  - containers.ports [reserved]
+  - containers.workingDir [reserved]
+- ephemeralContainers [reserved]
+- imagePullSecrets [alternative exists]
+- initContainers [reserved]
+- tolerations [alternative exists]
+
+Rules:
+
+If you are setting a value of a field in the podSpec or podSpec.containers[0],
```

Comment:
This is a pretty hard limitation, imagine that a user only wants to mount an additional volume, how do we easily enable this use-case?